portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```
@@ -89,18 +89,22 @@ private String getParameter(HttpServletRequest request, String comment, String d
     return value;
   }
 
-  private int getIntParameter(HttpServletRequest request, HttpServletResponse response, String comment_bound, int defaultValue) throws IOException {
-    int value = defaultValue;
+  private int getUnsignedIntParameter(HttpServletRequest request, HttpServletResponse response, String parameter_name, int defaultValue) throws IOException {
+    int value = 0;
 
     try {
-        value = Integer.parseInt(request.getParameter(comment_bound));
+        value = Integer.parseInt(request.getParameter(parameter_name));
     }
     catch (NumberFormatException e) {
-        response.sendError(400, "comment-bound must be a valid integer");
+        response.sendError(400, "paramater-name must be a valid integer");
     }
 
     if (value <= 0) {
-        return defaultValue;
+        if (defaultValue > 0) {
```

If the intent is that this function should always return zero or greater you might want to throw an error first thing if `defaultValue` < 0.